### PR TITLE
1757 mobx TerriaMap new routing

### PR DESCRIPTION
**Pull request by soyarsauce**

### CODE REVIEW

File: buildprocess/generate-init-routes.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const getIdFromItem = item => item.id;
+
+
+const getRoutes = (topCatalogItem, currentRoute = rootGroupName) =>
+  topCatalogItem.reduce((acc, catalogItem) => {
+    const nameForCurrentItem = getNameFromItem(catalogItem);
+    // Id is used for uniqueid, so disregard any pathing
+    const idFromItem = getIdFromItem(catalogItem);
+    const finalRoute = idFromItem || `${currentRoute}/${nameForCurrentItem}`;
+    const items = catalogItem.items && getRoutes(catalogItem.items, finalRoute);
```
Comment (na9da) [inline buildprocess/generate-init-routes.js:20]: This should be `catalog.members` following the `v8` schema.

Comment (soyarsauce) [reply buildprocess/generate-init-routes.js:20]: Ah yes, good pickup!

**Comment by soyarsauce** [comment]
CI passes if you use the right veersion fo terriajs
https://github.com/TerriaJS/TerriaMap/pull/349/commits

**Review by na9da — APPROVED** [review]

**Comment by soyarsauce** [comment]
Prerendering is opt-in for now

**Pull request merged**